### PR TITLE
Translation of Static Type Checking

### PR DESCRIPTION
**Pull request by vburlai**
Translation of static type checking

### CODE REVIEW

**Comment by vburlai** [comment]
Hi @klymenkoo 
Please review.

File: content/docs/static-type-checking.md [hunk 1]
```diff
@@ -312,24 +312,24 @@ declare module 'querystring' {
...
+[Reason](https://reasonml.github.io/uk/) – це не нова мова а новий синтаксис та інструментарій, працюючі на випробуваній в боях мові [OCaml](https://ocaml.org/). Reason надає OCaml знайомий синтаксис пристосований для програмістів JavaScript та використовує знайомий всім робочий процес NPM/Yarn.
 
-Reason is developed at Facebook, and is used in some of its products like Messenger. It is still somewhat experimental but it has [dedicated React bindings](https://reasonml.github.io/reason-react/) maintained by Facebook and a [vibrant community](https://reasonml.github.io/docs/en/community.html).
+Reason розробляється компанією Facebook і використовується в деяких її продуктах, таких як Messenger. Він поки є дещо експериментальною технологією, але для нього існують [спеціалізовані прив’язки з React](https://reasonml.github.io/reason-react/ru/), що підтримуються Facebook та [активна спільнота](https://reasonml.github.io/docs/uk/community.html).
```
Comment (vburlai) [inline content/docs/static-type-checking.md]: I'm using the link to Russian `reason-react` site here https://reasonml.github.io/reason-react/ru/. Is it okay or should I keep the original?

Comment (klymenkoo) [reply content/docs/static-type-checking.md]: It looks far from being completed, so keep the original link 👍

File: content/docs/static-type-checking.md [hunk 2]
```diff
@@ -1,44 +1,44 @@
...
 id: static-type-checking
-title: Static Type Checking
+title: Статична типізація
 permalink: docs/static-type-checking.html
 prev: typechecking-with-proptypes.html
 next: refs-and-the-dom.html
 ---
 
-Static type checkers like [Flow](https://flow.org/) and [TypeScript](https://www.typescriptlang.org/) identify certain types of problems before you even run your code. They can also improve developer workflow by adding features like auto-completion. For this reason, we recommend using Flow or TypeScript instead of `PropTypes` for larger code bases.
+Статичні типізатори, такі як [Flow](https://flow.org/) і [TypeScript](https://www.typescriptlang.org/), розпізнають певні типи проблем навіть до запуску вашого коду. Вони також можуть покращити робочий процес розробника додаючи функції, такі як автозавершення З цієї причини для великих кодових баз ми рекомендуємо використовувати Flow або TypeScript замість `PropTypes`.
```
Comment (klymenkoo) [inline content/docs/static-type-checking.md]:
```suggestion
Статичні типізатори, такі як [Flow](https://flow.org/) і [TypeScript](https://www.typescriptlang.org/), розпізнають певні типи проблем навіть до запуску вашого коду. Вони також можуть покращити робочий процес розробника додаючи функції, такі як автозавершення. З цієї причини для великих кодових баз ми рекомендуємо використовувати Flow або TypeScript замість `PropTypes`.
```

File: content/docs/static-type-checking.md [hunk 3]
```diff
@@ -1,44 +1,44 @@
...
 
 ## Flow {#flow}
 
-[Flow](https://flow.org/) is a static type checker for your JavaScript code. It is developed at Facebook and is often used with React. It lets you annotate the variables, functions, and React components with a special type syntax, and catch mistakes early. You can read an [introduction to Flow](https://flow.org/en/docs/getting-started/) to learn its basics.
+[Flow](https://flow.org/) — це статичний типізатор для вашого JavaScript-коду. Він розроблюється Facebook і часто використовується разом з React. Він дозволяє вам анотувати змінні, функції та React-компоненти спеціальним синтаксисом типу і рано піймати помилки. Щоб вивчити його основи ви можете прочитати [введення у Flow](https://flow.org/en/docs/getting-started/).
```
Comment (klymenkoo) [inline content/docs/static-type-checking.md]:
```suggestion
[Flow](https://flow.org/) — це статичний типізатор для вашого JavaScript-коду. Він розроблюється у Facebook і часто використовується разом з React. Він дозволяє вам анотувати змінні, функції та React-компоненти спеціальним синтаксисом типу і рано ловити помилки. Щоб вивчити його основи ви можете прочитати [введення у Flow](https://flow.org/en/docs/getting-started/).
```

File: content/docs/static-type-checking.md [hunk 4]
```diff
@@ -1,44 +1,44 @@
...
 
 ```bash
 npm install --save-dev flow-bin
 ```
 
-This command installs the latest version of Flow into your project.
+Ця команда встановить останню версію Flow у ваш проект.
 
-Now, add `flow` to the `"scripts"` section of your `package.json` to be able to use this from the terminal:
+Тепер додайте `flow` у відділення `"scripts"` вашого `package.json` щоб ви змогли використовувати його з терміналу:
```
Comment (klymenkoo) [inline content/docs/static-type-checking.md]:
```suggestion
Тепер додайте `flow` у відділення `"scripts"` вашого `package.json`, щоб ви змогли використовувати його з терміналу:
```

File: content/docs/static-type-checking.md [hunk 5]
```diff
@@ -51,53 +51,53 @@ Now, add `flow` to the `"scripts"` section of your `package.json` to be able to
...
+>Примітка:
 >
->These instructions are *not* for Create React App users. Even though Create React App uses Babel under the hood, it is already configured to understand Flow. Only follow this step if you *don't* use Create React App.
+>Ці інструкції *не* для користувачів Create React App. Незважаючи на те, що Create React App використовує Babel в реалізації, він вже налаштований на розуміння Flow. Виконайте це крок лише коли ви *не* використовуєте Create React App.
 
-If you manually configured Babel for your project, you will need to install a special preset for Flow.
+Якщо ви налаштували Babel для вашого проекту вручну, то вам потрібно буде встановити спеціальний пресет для Flow.
 
-If you use Yarn, run:
+Якщо ви користуєтесь Yarn то запустіть:
```
Comment (klymenkoo) [inline content/docs/static-type-checking.md]:
```suggestion
Якщо ви користуєтесь Yarn, то запустіть:
```

File: content/docs/static-type-checking.md [hunk 6]
```diff
@@ -51,53 +51,53 @@ Now, add `flow` to the `"scripts"` section of your `package.json` to be able to
...
 
 ```bash
 yarn add --dev babel-preset-flow
 ```
 
-If you use npm, run:
+Якщо ви користуєтесь npm то запустіть:
```
Comment (klymenkoo) [inline content/docs/static-type-checking.md]:
```suggestion
Якщо ви користуєтесь npm, то запустіть:
```

File: content/docs/static-type-checking.md [hunk 7]
```diff
@@ -108,101 +108,101 @@ Then add the `flow` preset to your [Babel configuration](https://babeljs.io/docs
...
 
-If you followed the instructions above, you should be able to run Flow for the first time.
+Якщо ви слідували наведеним вище інструкціям, то ви повинні мати можливість вперше запустити Flow.
 
 ```bash
 yarn flow
 ```
 
-If you use npm, run:
+Якщо ви користуєтесь npm то запустіть:
```
Comment (klymenkoo) [inline content/docs/static-type-checking.md]:
```suggestion
Якщо ви користуєтесь npm, то запустіть:
```

File: content/docs/static-type-checking.md [hunk 8]
```diff
@@ -108,101 +108,101 @@ Then add the `flow` preset to your [Babel configuration](https://babeljs.io/docs
...
 
-By default, Flow only checks the files that include this annotation:
+За замовчуванням Flow перевіряє лише файли, що включають таку анотацію:
 
 ```js
 // @flow
 ```
 
-Typically it is placed at the top of a file. Try adding it to some files in your project and run `yarn flow` or `npm run flow` to see if Flow already found any issues.
+Зазвичай вона розміщується на початку файлу. Спробуйте додати її до деяких файлів у вашому проекті і запустити `yarn flow` чи `npm run flow` щоб побачити чи Flow вже знайшов якісь проблеми.
```
Comment (klymenkoo) [inline content/docs/static-type-checking.md]:
```suggestion
Зазвичай вона розміщується на початку файлу. Спробуйте додати її до деяких файлів у вашому проекті і запустити `yarn flow` чи `npm run flow`, щоб побачити чи Flow вже знайшов якісь проблеми.
```

File: content/docs/static-type-checking.md [hunk 9]
```diff
@@ -108,101 +108,101 @@ Then add the `flow` preset to your [Babel configuration](https://babeljs.io/docs
...
 
-There is also [an option](https://flow.org/en/docs/config/options/#toc-all-boolean) to force Flow to check *all* files regardless of the annotation. This can be too noisy for existing projects, but is reasonable for a new project if you want to fully type it with Flow.
+Також існує [опція](https://flow.org/en/docs/config/options/#toc-all-boolean), що дозволяє примусити Flow перевіряти *всі* файли незалежно від наявності вищевказаної анотації. Це може бути надто шумним для існуючих проектів, але є прийнятним для нових проектів якщо ви хочете повністю типізувати його з Flow.
```
Comment (klymenkoo) [inline content/docs/static-type-checking.md]:
```suggestion
Також існує [опція](https://flow.org/en/docs/config/options/#toc-all-boolean), що дозволяє примусити Flow перевіряти *всі* файли незалежно від наявності вищевказаної анотації. Якщо ви додаєте Flow в існуючий проект, то може виникнути багато конфліктів, але якщо ви починаєте з нуля, то це непогана можливість зробити це.
```

File: content/docs/static-type-checking.md [hunk 10]
```diff
@@ -108,101 +108,101 @@ Then add the `flow` preset to your [Babel configuration](https://babeljs.io/docs
...
 
-Now you're all set! We recommend to check out the following resources to learn more about Flow:
+Тепер у вас все налаштовано! Ми рекомендуємо ознайомитись з наступними ресурсами щоб дізнатися більше про Flow:
```
Comment (klymenkoo) [inline content/docs/static-type-checking.md]:
```suggestion
Тепер у вас все налаштовано! Ми рекомендуємо ознайомитись з наступними ресурсами, щоб дізнатися більше про Flow:
```

File: content/docs/static-type-checking.md [hunk 11]
```diff
@@ -259,40 +259,40 @@ We'll address this in two steps:
...
-Great! Now when we run our build script the compiler will output the generated javascript to the `build` folder. The [TypeScript React Starter](https://github.com/Microsoft/TypeScript-React-Starter/blob/master/tsconfig.json) provides a `tsconfig.json` with a good set of rules to get you started.
+Чудово! Тепер при запуску скрипту збірки компілятор виведе згенеровані файли JavaScript в папку `build`. [TypeScript React Starter](https://github.com/Microsoft/TypeScript-React-Starter/blob/master/tsconfig.json) надає `tsconfig.json` з гарним набором правил для початку.
 
-Generally, you don't want to keep the generated javascript in your source control, so be sure to add the build folder to your `.gitignore`.
+Зазвичай згенерований JavaScript-код не варто зберігати в системі керування версіями, тому не забудьте додати папку `build` до вашого `.gitignore`.
 
-### File extensions {#file-extensions}
-In React, you most likely write your components in a `.js` file. In TypeScript we have 2 file extensions:
+### Розширення файлів {#file-extensions}
+При роботі з React ви, скоріше за все, створювали свої компоненти у файлах `.js`. В TypeScript ми маємо два різних файлових розширення:
```
Comment (klymenkoo) [inline content/docs/static-type-checking.md]:
```suggestion
При роботі з React ви, скоріше за все, створювали свої компоненти у файлах `.js`. У TypeScript ми маємо два різних файлових розширення:
```

File: content/docs/static-type-checking.md [hunk 12]
```diff
@@ -259,40 +259,40 @@ We'll address this in two steps:
...
 
-If you followed the instructions above, you should be able to run TypeScript for the first time.
+Якщо ви слідували інструкціям вище, то ви повинні мати можливість вперше запустити TypeScript.
 
 ```bash
 yarn build
 ```
 
-If you use npm, run:
+Якщо ви користуєтесь npm то запустіть:
```
Comment (klymenkoo) [inline content/docs/static-type-checking.md]:
```suggestion
Якщо ви користуєтесь npm, то запустіть:
```

File: content/docs/static-type-checking.md [hunk 13]
```diff
@@ -312,24 +312,24 @@ declare module 'querystring' {
...
-* [TypeScript Documentation: React and Webpack](https://www.typescriptlang.org/docs/handbook/react-&-webpack.html)
+* [Документація TypeScript: базові типи](https://www.typescriptlang.org/docs/handbook/basic-types.html)
+* [Документація TypeScript: міграція з JavaScript](https://www.typescriptlang.org/docs/handbook/migrating-from-javascript.html)
+* [Документація TypeScript: React та Webpack](https://www.typescriptlang.org/docs/handbook/react-&-webpack.html)
 
 ## Reason {#reason}
 
-[Reason](https://reasonml.github.io/) is not a new language; it's a new syntax and toolchain powered by the battle-tested language, [OCaml](https://ocaml.org/). Reason gives OCaml a familiar syntax geared toward JavaScript programmers, and caters to the existing NPM/Yarn workflow folks already know.
+[Reason](https://reasonml.github.io/uk/) – це не нова мова а новий синтаксис та інструментарій, працюючі на випробуваній в боях мові [OCaml](https://ocaml.org/). Reason надає OCaml знайомий синтаксис пристосований для програмістів JavaScript та використовує знайомий всім робочий процес NPM/Yarn.
```
Comment (klymenkoo) [inline content/docs/static-type-checking.md]:
```suggestion
[Reason](https://reasonml.github.io/uk/) – це не нова мова, а новий синтаксис та інструментарій, працюючі на перевіреній часом мові [OCaml](https://ocaml.org/). Reason надає OCaml знайомий синтаксис пристосований для програмістів JavaScript та використовує знайомий всім робочий процес NPM/Yarn.
```

File: content/docs/static-type-checking.md [hunk 14]
```diff
@@ -312,24 +312,24 @@ declare module 'querystring' {
...
 
-Reason is developed at Facebook, and is used in some of its products like Messenger. It is still somewhat experimental but it has [dedicated React bindings](https://reasonml.github.io/reason-react/) maintained by Facebook and a [vibrant community](https://reasonml.github.io/docs/en/community.html).
+Reason розробляється компанією Facebook і використовується в деяких її продуктах, таких як Messenger. Він поки є дещо експериментальною технологією, але для нього існують [спеціалізовані прив’язки з React](https://reasonml.github.io/reason-react/), що підтримуються Facebook та [активна спільнота](https://reasonml.github.io/docs/uk/community.html).
```
Comment (klymenkoo) [inline content/docs/static-type-checking.md]:
```suggestion
Reason розробляється компанією Facebook і використовується в деяких її продуктах, таких як Messenger. Він поки є дещо експериментальною технологією, але для нього існують [спеціалізовані прив’язки з React](https://reasonml.github.io/reason-react/), що підтримуються Facebook та [активною спільнотою](https://reasonml.github.io/docs/uk/community.html).
```

File: content/docs/static-type-checking.md [hunk 15]
```diff
@@ -312,24 +312,24 @@ declare module 'querystring' {
...
 
 ## Kotlin {#kotlin}
 
-[Kotlin](https://kotlinlang.org/) is a statically typed language developed by JetBrains. Its target platforms include the JVM, Android, LLVM, and [JavaScript](https://kotlinlang.org/docs/reference/js-overview.html). 
+[Kotlin](https://kotlinlang.org/) – це статично типізована мова розроблена JetBrains. Її цільові платформи включають JVM, Android, LLVM, та [JavaScript](https://kotlinlang.org/docs/reference/js-overview.html). 
 
-JetBrains develops and maintains several tools specifically for the React community: [React bindings](https://github.com/JetBrains/kotlin-wrappers) as well as [Create React Kotlin App](https://github.com/JetBrains/create-react-kotlin-app). The latter helps you start building React apps with Kotlin with no build configuration.
+JetBrains розробляє та підтримує декілька інструментів спеціально для спільноти користувачів React: [React-прив’язки](https://github.com/JetBrains/kotlin-wrappers) а також [Create React Kotlin App](https://github.com/JetBrains/create-react-kotlin-app). Останній допомагає почати будувати React-додатки з Kotlin без потреби в налаштуванні збірки.
```
Comment (klymenkoo) [inline content/docs/static-type-checking.md]:
```suggestion
JetBrains розробляє та підтримує декілька інструментів спеціально для спільноти користувачів React: [React-прив’язки](https://github.com/JetBrains/kotlin-wrappers) а також [Create React Kotlin App](https://github.com/JetBrains/create-react-kotlin-app). Останній допомагає почати будувати React-додатки з Kotlin без потреби у налаштуванні збірки.
```

File: content/docs/static-type-checking.md [hunk 16]
```diff
@@ -312,24 +312,24 @@ declare module 'querystring' {
...
 
-## Other Languages {#other-languages}
+## Інші мови {#other-languages}
 
-Note there are other statically typed languages that compile to JavaScript and are thus React compatible. For example, [F#/Fable](https://fable.io/) with [elmish-react](https://elmish.github.io/react). Check out their respective sites for more information, and feel free to add more statically typed languages that work with React to this page!
+Зверніть увагу, що існують інші статично типізовані мови які компілюються в JavaScript і таким чином сумісні з React. Наприклад, [F#/Fable](https://fable.io/) з [elmish-react](https://elmish.github.io/react). Ознайомтесь з їх відповідними сайтами щоб дізнатися більше інформації, також не вагаючись додавайте до цієї сторінки інші статично типізовані мови, що працюють з React!
```
Comment (klymenkoo) [inline content/docs/static-type-checking.md]:
```suggestion
Зверніть увагу, що існують інші статично типізовані мови які компілюються у JavaScript і таким чином сумісні з React. Наприклад, [F#/Fable](https://fable.io/) з [elmish-react](https://elmish.github.io/react). Ознайомтесь з їх відповідними сайтами щоб дізнатися більше інформації, також не вагайтеся додовати до цієї сторінки інші статично-типізовані мови, що працюють з React!
```

File: content/docs/static-type-checking.md (hunk 13 above)
Comment (vburlai) [reply content/docs/static-type-checking.md]: 👍

File: content/docs/static-type-checking.md [hunk 17]
```diff
@@ -320,16 +320,16 @@ declare module 'querystring' {
...
+JetBrains розробляє та підтримує декілька інструментів спеціально для спільноти користувачів React: [React-прив’язки](https://github.com/JetBrains/kotlin-wrappers) а також [Create React Kotlin App](https://github.com/JetBrains/create-react-kotlin-app). Останній допомагає почати будувати React-додатки з Kotlin без потреби у налаштуванні збірки.
 
 ## Інші мови {#other-languages}
 
-Зверніть увагу, що існують інші статично типізовані мови які компілюються в JavaScript і таким чином сумісні з React. Наприклад, [F#/Fable](https://fable.io/) з [elmish-react](https://elmish.github.io/react). Ознайомтесь з їх відповідними сайтами щоб дізнатися більше інформації, також не вагаючись додавайте до цієї сторінки інші статично типізовані мови, що працюють з React!
+Зверніть увагу, що існують інші статично типізовані мови які компілюються у JavaScript і таким чином сумісні з React. Наприклад, [F#/Fable](https://fable.io/) з [elmish-react](https://elmish.github.io/react). Ознайомтесь з їх відповідними сайтами щоб дізнатися більше інформації, також не вагайтеся додовати до цієї сторінки інші статично-типізовані мови, що працюють з React!
```
Comment (klymenkoo) [inline content/docs/static-type-checking.md]: Додавати

**Pull request merged**